test/functional/users.js

```diff
@@ -943,28 +946,24 @@ describe("UsersController", function() {
           data.users.frontendPreferences['custom.domain'].customProperty.should.equal('someWeirdValue')
         }
 
-        // Fourth, only update some sub-objects (frontendPreferences should get deep-merged)
+        // Fourth, only update some sub-objects (frontendPreferences should get shallow-merged)
         {
           await funcTestHelper.updateUserAsync({ user, authToken }, { frontendPreferences: newPrefs })
 
           let response = await funcTestHelper.whoami(authToken)
           response.status.should.eql(200)
 
           let data = await response.json()
+          data.should.have.deep.property('users.frontendPreferences')
           // another.client
-          data.should.have.deep.property('users.frontendPreferences.another\\.client')
-          data.users.frontendPreferences['another.client'].should.have.property('funnyProperty')
-          data.users.frontendPreferences['another.client'].funnyProperty.should.equal('withFunnyValue')
+          data.users.frontendPreferences.should.have.property('another.client')
+          data.users.frontendPreferences['another.client'].should.be.deep.equal(newPrefs['another.client'])
           // net.freefeed
           data.users.frontendPreferences.should.have.property('net.freefeed')
-          data.users.frontendPreferences['net.freefeed'].should.have.deep.property('screenName.displayOption')
-          data.users.frontendPreferences['net.freefeed'].screenName.displayOption.should.equal(2)
-          data.users.frontendPreferences['net.freefeed'].should.have.deep.property('screenName.useYou')
-          data.users.frontendPreferences['net.freefeed'].screenName.useYou.should.equal(false)
+          data.users.frontendPreferences['net.freefeed'].should.be.deep.equal(newPrefs['net.freefeed'])
           // custom domain
           data.users.frontendPreferences.should.have.property('custom.domain')
-          data.users.frontendPreferences['custom.domain'].should.have.property('customProperty')
-          data.users.frontendPreferences['custom.domain'].customProperty.should.equal('someWeirdValue')
+          data.users.frontendPreferences['custom.domain'].should.be.deep.equal(newPrefs['custom.domain'])
```

Comment:
Actually, maybe we should leave "custom.domain" its old role, and just introduce a new domain to test shallow merging.